File: examples/Parity.swift
import Foundation

struct ParityRow: Codable {
    let seed: String
    let attr: String
    let expected: UInt32
}

struct ParityFile: Codable {
    let bucket: [ParityRow]
}

@main
struct ParityMain {
    static func main() throws {
        let root: URL
        if CommandLine.arguments.count > 1 {
            root = URL(fileURLWithPath: CommandLine.arguments[1])
        } else {
            root = URL(fileURLWithPath: FileManager.default.currentDirectoryPath)
        }
        let data = try Data(contentsOf: root.appendingPathComponent("schemas/parity-fixture.json"))
        let doc = try JSONDecoder().decode(ParityFile.self, from: data)
        var failed = 0
        for row in doc.bucket {
            let got = bucket(seed: row.seed, attr: row.attr)
            let ok = got == row.expected
            if !ok { failed += 1 }
            print("swift \(ok ? "ok" : "FAIL") seed=\(row.seed.debugDescription) attr=\(row.attr.debugDescription) got=\(got) expected=\(row.expected)")
        }
        if failed > 0 {
            fputs("swift parity failures: \(failed)\n", stderr)
            exit(1)
        }
        print("swift parity tuples=\(doc.bucket.count) status=ok")
    }
}

File: examples/parity.mjs
import { readFileSync } from "node:fs";
import { dirname, join } from "node:path";
import { fileURLToPath } from "node:url";
import { bucket } from "../ts/evaluator.ts";

const root = join(dirname(fileURLToPath(import.meta.url)), "..");
const doc = JSON.parse(
  readFileSync(join(root, "schemas", "parity-fixture.json"), "utf8"),
);
let failed = 0;
for (const row of doc.bucket) {
  const got = bucket(row.seed, row.attr);
  const ok = got === row.expected;
  if (!ok) failed += 1;
  console.log(
    `ts ${ok ? "ok" : "FAIL"} seed=${JSON.stringify(row.seed)} attr=${JSON.stringify(row.attr)} got=${got} expected=${row.expected}`,
  );
}
if (failed > 0) {
  console.error(`ts parity failures: ${failed}`);
  process.exit(1);
}
console.log(`ts parity tuples=${doc.bucket.length} status=ok`);
